src/createcluster.ts

```diff
@@ -0,0 +1,543 @@
+'use strict';
+
+import { TextDocumentContentProvider, Uri, EventEmitter, Event, ProviderResult, CancellationToken } from 'vscode';
+import { Shell } from './shell';
+import { FS } from './fs';
+import { Advanceable, Errorable, UIRequest, StageData, OperationState, OperationMap, advanceUri as wizardAdvanceUri, selectionChangedScript as wizardSelectionChangedScript, selectionChangedScriptMulti as wizardSelectionChangedScriptMulti, script, waitScript, extend, ControlMapping, styles } from './wizard';
+import { Context, getSubscriptionList, loginAsync, ServiceLocation, Locations } from './azure';
+import { error } from 'util';
+
+export const uriScheme : string = "k8screatecluster";
+
+export function operationUri(operationId: string) : Uri {
+    return Uri.parse(`${uriScheme}://operations/${operationId}`);
+}
+
+export function uiProvider(fs: FS, shell: Shell) : TextDocumentContentProvider & Advanceable {
+    return new UIProvider(fs, shell);
+}
```

Comment:
If we want this to be for any type of kubernetes cloud/on-prem then we probably need to externalize the rendering function per type.

Reply:
There's still a single entry point where we prompt for what kind of cluster to create.  We can do a better job of teasing apart the generic wizard logic, the common prompts, and the Azure-specific pages, but until we or a third party embarks on a new cluster type that would be pretty speculative.  I vote to leave for now, knowing that it will need refactoring later but that we will have better information about commonalities and differences then.